### PR TITLE
Update hyper to 1.3.1 and reqwest to 0.12.5.

### PR DESCRIPTION
**Pull request by partim**
This PR updates hyper to 1.3.1 and reqwest to 0.12.4 and rearranges the HTTP server to the changes in hyper 1.0.

### CODE REVIEW

File: src/daemon/http/mod.rs [hunk 1]
```diff
@@ -433,67 +452,23 @@ impl Request {
...
 
-    /// See hyper::body::to_bytes
-    ///
-    /// Here we want to limit the bytes consumed to a maximum. So, the
-    /// code below is adapted from the method in the hyper crate.
     pub async fn read_bytes(self, limit: u64) -> Result<Bytes, Error> {
-        let body = self.request.into_body();
-
-        futures_util::pin_mut!(body);
+        // We’re going to cheat a bit. If we know the body is too big from
```
Comment (ximon18) [inline src/daemon/http/mod.rs:456]: It's not clear to me if this code, or even the old code, actually prevents reading too much data into memory from the network.

File: src/daemon/http/tls.rs [hunk 2]
```diff
@@ -1,322 +1,380 @@
...
-        let mut config = ServerConfig::builder()
-            .with_safe_default_cipher_suites()
-            .with_safe_default_kx_groups()
-            .with_safe_default_protocol_versions()
-            .map_err(TlsConfigError::Rustls)?
-            .with_no_client_auth()
-            .with_single_cert(cert, tokio_rustls::rustls::PrivateKey(key))
-            .map_err(TlsConfigError::Rustls)?;
-
-        // See: https://wiki.wireshark.org/TLS#tls-decryption
```
Comment (ximon18) [inline src/daemon/http/tls.rs:185]: Is there a reason that this comment wasn't carried over to the new code?

File: src/daemon/http/tls.rs [hunk 3]
```diff
@@ -1,322 +1,380 @@
...
+        Accept { #[pin] fut: Accept<TcpStream> },
+
+        /// We have a working TLS stream.
+        Stream { #[pin] fut: TlsStream<TcpStream> },
+
+        /// TLS handshake has failed.
+        ///
+        /// Because hyper still wants to do a clean flush and shutdown, we
+        /// need to still work in this state. For read and write, we just
+        /// keep returning the clean shutdown indiciation of zero length
```
Comment (ximon18) [inline src/daemon/http/tls.rs:129]:
```suggestion
        /// keep returning the clean shutdown indication of zero length
```

File: src/daemon/http/tls.rs [hunk 4]
```diff
@@ -1,322 +1,380 @@
...
-
-impl std::fmt::Display for TlsConfigError {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            TlsConfigError::Io(err) => err.fmt(f),
-            TlsConfigError::Pkcs8ParseError => write!(f, "pkcs8 parse error"),
-            TlsConfigError::RsaParseError => write!(f, "rsa parse error"),
-            TlsConfigError::EmptyKey => write!(f, "key contains no private key"),
-            TlsConfigError::Rustls(err) => write!(f, "rustls error: {}", err),
+/// Reads the first private key from the given PEM file.
```
Comment (ximon18) [inline src/daemon/http/tls.rs:60]: "First" here conflicts with "if there isn't exactly one private key" below.

File: src/daemon/http/tls.rs [hunk 5]
```diff
@@ -1,322 +1,380 @@
...
+///
+/// The key may be a PKCS#1 RSA private key, a PKCS#8 private key, or a
+/// SEC1 encoded EC private key. All other PEM items are ignored.
+///
+/// Errors out if opening or reading the file fails or if there isn’t exactly
```
Comment (ximon18) [inline src/daemon/http/tls.rs:65]: The while loop seems to try to the next item if the current one is not supported... so this comment is incorrect I think?

File: src/daemon/http/tls.rs [hunk 6]
```diff
@@ -1,322 +1,380 @@
...
+                        Poll::Ready(Err(err))
+                    }
                 }
             }
-        };
-
-        let mut config = ServerConfig::builder()
-            .with_safe_default_cipher_suites()
-            .with_safe_default_kx_groups()
-            .with_safe_default_protocol_versions()
```
Comment (ximon18) [inline src/daemon/http/tls.rs:179]: Is there a reason that `.with_safe_default_protocol_versions()` wasn't carried over to the new code?

**Review by ximon18 — APPROVED** [review]
Added some comments but looks mostly fine.

File: src/daemon/http/mod.rs (hunk 1 above)
Comment (partim) [reply src/daemon/http/mod.rs:456]: My understanding is that Hyper will only start reading the body when you explicitly tell it to. The first test here basically looks at the Content-Length header and rejects the request if that’s too large before even beginning to read the body. The new `Limited` thing is only relevant for chunked transfers.

File: src/daemon/http/tls.rs (hunk 2 above)
Comment (partim) [reply src/daemon/http/tls.rs:185]: Nope. I added it back in again.

File: src/daemon/http/tls.rs (hunk 5 above)
Comment (partim) [reply src/daemon/http/tls.rs:65]: Changed it to “… if there isn’t exactly one _recognized_ private key in the file.”

File: src/daemon/http/tls.rs (hunk 6 above)
Comment (partim) [reply src/daemon/http/tls.rs:179]: `ServerConfig::builder` now returns a builder with all these things already set.

**Pull request merged**